portal-bridge/src/types/state/storage/account_db.rs

```diff
@@ -0,0 +1,99 @@
+use std::sync::Arc;
+
+use alloy_primitives::{keccak256, Address, B256};
+use eth_trie::DB;
+use rocksdb::DB as RocksDB;
+
+use super::error::EVMError;
+
+static NULL_RLP_STATIC: [u8; 1] = [0x80; 1];
+
+#[derive(Debug, Clone)]
+pub struct AccountDB {
+    pub address_hash: B256,
+    /// storage trie
+    pub db: Arc<RocksDB>,
+}
+
+impl AccountDB {
+    pub fn new(address: Address, db: Arc<RocksDB>) -> Self {
+        Self {
+            address_hash: keccak256(address),
+            db,
+        }
+    }
+
+    fn combine_key(addr_hash: &[u8], key: &[u8]) -> Vec<u8> {
```

Comment:
You could have made this:

```rust
fn get_db_key(&self, key: &[u8]) -> Vec<u8> {
  [self.address_hash.as_slice(), key].concat()
}
```

Then later, you would do: `self.db.get(self.get_db_key(key))`, `self.db.put(self.get_db_key(key), ...)` and `self.db.delete(self.get_db_key(key))`

Reply:
I will address this in my dao pr as well